sycl/include/CL/sycl/types.hpp

```diff
@@ -729,7 +728,12 @@ template <typename Type, int NumElements> class vec {
 #ifdef __SYCL_DEVICE_ONLY__
 #define __SYCL_RELLOGOP(RELLOGOP)                                              \
   vec<rel_t, NumElements> operator RELLOGOP(const vec &Rhs) const {            \
-    return vec<rel_t, NumElements>{m_Data RELLOGOP vector_t(Rhs)};             \
+    auto Ret =                                                                 \
+        vec<rel_t, NumElements>((typename vec<rel_t, NumElements>::vector_t)(  \
+            m_Data RELLOGOP Rhs.m_Data));                                      \
+    if (NumElements == 1) /*Scalar 0/1 logic was applied, invert*/             \
+      Ret *= -1;                                                               \
```

Comment:
Why not `Ret = -Ret` by the way?

Reply:
I'm not sure if it matters much.

Reply:
@keryell 
the class vector has no unary minus operator.

Reply:
That seems like a SYCL spec bug... If I understand correctly, I do not want to have a vector of multipliers on FPGA for example just because we do not have a unary minus operator... Probably a peephole optimization pass will replace the *-1 by - in the back-end, but we never know...